### PR TITLE
Cart and wishlist

### PR DESCRIPTION
**Pull request by Ash-D23**
Added Cart and Wishlist Pages
Added Cart and WishList Context
Added add/remove to cart and wishlist functionalities into products

sss
sss1

### CODE REVIEW

File: src/Context/CartContext/CartContext.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        let itemincart = checkitemincart(item._id)
+  
+        if(itemincart){
+          // if item in cart filter items and add 1 quantity and append
+          increasequantity(item)
+        }else{
+          // if item not in cart set cart with item and quantity 1
+          try{
+            let result = await axios.post('/api/user/cart', { product: { ...item } } , config)
+            console.log(result)
```
Comment (manismk) [inline src/Context/CartContext/CartContext.js]: can remove the console logs

**Review by manismk — APPROVED** [review]
Suggestion: you can use barrel exports.

Looks good. Code is easy to understand

File: src/Context/CartContext/CartContext.js (hunk 1 above)
Comment (Ash-D23) [reply src/Context/CartContext/CartContext.js]: Sure will do that

File: src/Components/CartProducts/CartProducts.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+function CartProducts() {
+
+  const {cartitems, cartloading} = useCart()
+  
+  return (
+      <div className="checkout__products">
+          { cartitems.length === 0 ? <p className="text--large margin--large">No items in Cart</p> : <h3 className="checkout__title margin--large">Items in Cart</h3> }
+          { cartloading && <Loader />}
+          {cartitems?.map((item)=>{
+            return <SingleCartProduct cartItem={item} />
```
Comment (shobit1337) [inline src/Components/CartProducts/CartProducts.js]: Use key attributes here.

File: src/Pages/WishlistPage/WishListPage.js [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
+function WishlistPage() {
+
+  const { wishlistitems } = useWishlist()
   return (
-    <div>WishListPage</div>
+    <div className="container--90">
+            <h2 className="text--center padding--large">Wishlist</h2>
+            <div className="product--cards container__flex--center container__flex--wrap">
+                {wishlistitems?.map((item)=> {
+                  return <SingleProduct product={item} wishlistproduct={true} />
```
Comment (shobit1337) [inline src/Pages/WishlistPage/WishListPage.js]: Always use key prop when rendering jsx with .map

File: src/Context/AuthContext/AuthContext.js [hunk 4]
```diff
@@ -24,9 +24,8 @@ const useAuth = () => {
         console.log(data)
```
Comment (shobit1337) [inline src/Context/AuthContext/AuthContext.js]: Remove there unnecessary, console.logs before pushing your code to GitHub.

**Review by shobit1337 — APPROVED** [review]
Everything looks good and working.

File: src/Pages/WishlistPage/WishListPage.js (hunk 3 above)
Comment (Ash-D23) [reply src/Pages/WishlistPage/WishListPage.js]: yes will add now, thanks

**Pull request merged**